src/module_wrap.h

```diff
@@ -38,6 +38,44 @@ enum ModulePhase : int {
   kEvaluationPhase = 2,
 };
 
+struct ModuleRequest : public MemoryRetainer {
+  std::string specifier;
+  std::vector<std::pair<std::string, std::string>> import_attributes;
+  ModulePhase phase;
+
+  ModuleRequest(
+      std::string specifier,
+      std::vector<std::pair<std::string, std::string>> import_attributes,
+      ModulePhase phase)
+      : specifier(specifier),
+        import_attributes(import_attributes),
+        phase(phase) {}
+
+  SET_MEMORY_INFO_NAME(ModuleRequest)
+  SET_SELF_SIZE(ModuleRequest)
+  void MemoryInfo(MemoryTracker* tracker) const override;
+
+  struct Hash {
+    std::size_t operator()(const ModuleRequest& request) const {
+      std::size_t h1 = std::hash<std::string>{}(request.specifier);
```

Comment:
For string hashes I think it can be computed as the module request is constructed with `v8::Name::GetIdentityHash()` which is just an inline property access, and store it into the wrapper. There is no need to hash it again over the content using `std::hash`.

Reply:
Thanks for suggestion! Updated, PTAL